operatorcontrolplane/v1alpha1/types_conditioncheck.go

```diff
@@ -0,0 +1,150 @@
+// Package v1alpha1 is an API version in the controlplane.operator.openshift.io group
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodNetworkConnectivityCheck
+type PodNetworkConnectivityCheck struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the source and target of the connetivity check
+	// +kubebuilder:validation:Required
+	// +required
+	Spec PodNetworkConnectivityCheckSpec `json:"spec"`
+
+	// Status contains the observed status of the connectivity check
+	// +optional
+	Status PodNetworkConnectivityCheckStatus `json:"status"`
+}
+
+type PodNetworkConnectivityCheckSpec struct {
+	// SourcePod names the pod from which the condition will be checked
+	// +kubebuilder:validation:Required
+	// +required
+	SourcePod string `json:"sourcePod"`
+
+	// EndpointAddress to check. A TCP address of the form host:port. Note that
+	// if host is a DNS name, then the check would fail if the DNS name cannot
+	// be resolved. Specify an IP address for host to bypass DNS name lookup.
+	// +kubebuilder:validation:Required
+	// +required
+	TargetEndpoint string `json:"targetEndpoint"`
+}
+
+// +k8s:deepcopy-gen=true
+type PodNetworkConnectivityCheckStatus struct {
+	// Successes contains logs successful check actions
+	// +optional
+	Successes []LogEntry `json:"successes"`
```

Comment:
the last n? who defines `n`?

Reply:
> the last n? who defines `n`?

the binary itself.  let's say 10  in that binary to start.  We do similar sorts of things for our operator revisions.